--- uboot-mizy.ro/cmd/boot.c	2017-02-08 10:11:06.000000000 +0700
+++ uboot-mizy.rw/cmd/boot.c	2017-03-17 09:56:12.821465515 +0700
@@ -62,6 +62,12 @@ U_BOOT_CMD(
 	""
 );
 
+U_BOOT_CMD(
+	reboot, 1, 0,	do_reset,
+	"Perform RESET of the CPU - is reset alias",
+	""
+);
+
 #ifdef CONFIG_CMD_POWEROFF
 U_BOOT_CMD(
 	poweroff, 1, 0,	do_poweroff,
@@ -69,3 +75,9 @@ U_BOOT_CMD(
 	""
 );
 #endif
+
+// ## hyphop ##
+// add extra cmd
+
+#include "sunxi_spi_spl.c"
+#include "felfix.c"
