--- uboot-mizy.ro/common/autoboot.c	2017-02-08 10:11:06.000000000 +0700
+++ uboot-mizy.rw/common/autoboot.c	2017-03-10 22:49:25.615520089 +0700
@@ -212,7 +212,12 @@ static int __abortboot(int bootdelay)
 {
 	int abort = 0;
 	unsigned long ts;
-
+	
+	int delta = 1;
+	if ( bootdelay > 9 ) {
+	    delta=10;
+	}
+	
 #ifdef CONFIG_MENUPROMPT
 	printf(CONFIG_MENUPROMPT);
 #else
@@ -229,7 +234,7 @@ static int __abortboot(int bootdelay)
 	}
 
 	while ((bootdelay > 0) && (!abort)) {
-		--bootdelay;
+		bootdelay = bootdelay - delta;
 		/* delay 1000 ms */
 		ts = get_timer(0);
 		do {
@@ -243,8 +248,8 @@ static int __abortboot(int bootdelay)
 # endif
 				break;
 			}
-			udelay(10000);
-		} while (!abort && get_timer(ts) < 1000);
+			udelay( 10000 / delta );
+		} while (!abort && get_timer(ts) < 1000 / delta );
 
 		printf("\b\b\b%2d ", bootdelay);
 	}
