diff -pruN uboot-mizy/cmd/source.c uboot-miZy/cmd/source.c
--- uboot-mizy/cmd/source.c	2017-02-08 10:11:06.000000000 +0700
+++ uboot-miZy/cmd/source.c	2017-02-01 21:16:39.000000000 +0700
@@ -43,7 +43,8 @@ source (ulong addr, const char *fit_unam
 	size_t		fit_len;
 #endif
 
-	verify = getenv_yesno ("verify");
+//	verify = getenv_yesno ("verify");
+	verify = 0;
 
 	buf = map_sysmem(addr, 0);
 	switch (genimg_get_format(buf)) {
@@ -52,32 +53,36 @@ source (ulong addr, const char *fit_unam
 		hdr = buf;
 
 		if (!image_check_magic (hdr)) {
-			puts ("Bad magic number\n");
-			return 1;
+//			puts ("Bad magic number\n");
+			printf("Bad magic number\n");
+//			return 1;
 		}
 
 		if (!image_check_hcrc (hdr)) {
-			puts ("Bad header crc\n");
-			return 1;
+//			puts ("Bad header crc\n");
+			printf("Bad header crc\n");
+//			return 1;
 		}
 
-		if (verify) {
+//		if (verify) {
 			if (!image_check_dcrc (hdr)) {
-				puts ("Bad data crc\n");
-				return 1;
+//				puts ("Bad data crc\n");
+				printf("Bad data crc\n");
+///				return 1;
 			}
-		}
+//		}
 
 		if (!image_check_type (hdr, IH_TYPE_SCRIPT)) {
-			puts ("Bad image type\n");
-			return 1;
+//			puts ("Bad image type\n");
+			printf("Bad image type\n");
+//			return 1;
 		}
 
 		/* get length of script */
 		data = (u32 *)image_get_data (hdr);
 
 		if ((len = uimage_to_cpu (*data)) == 0) {
-			puts ("Empty Script\n");
+			printf ("Empty Script\n");
 			return 1;
 		}
 
